src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2356,4 +2363,27 @@ public void storeSpecialFieldsPreferences(SpecialFieldsPreferences specialFields
         putBoolean(AUTOSYNCSPECIALFIELDSTOKEYWORDS, specialFieldsPreferences.getAutoSyncSpecialFieldsToKeyWords());
         putBoolean(SERIALIZESPECIALFIELDS, specialFieldsPreferences.getSerializeSpecialFields());
     }
+
+    @Override
+    public SearchPreferences getSearchPreferences() {
+        SearchDisplayMode searchDisplayMode;
+        try {
+            searchDisplayMode = SearchDisplayMode.valueOf(get(SEARCH_DISPLAY_MODE));
+        } catch (IllegalArgumentException ex) {
+            // Should only occur when the searchmode is set directly via preferences.put and the enum was not used
+            searchDisplayMode = SearchDisplayMode.valueOf((String) defaults.get(SEARCH_DISPLAY_MODE));
```

Comment:
The get() Method already returns the default if no value:
```
 public String get(String key) {
        return prefs.get(key, (String) defaults.get(key));
    }
```

Reply:
Good catch. I just copied the logic from the former searchpreferences.class. Fixing the missing consistency is one of the big projects in the future...

Reply:
This was the only issue I found so far,the rest looks good!

Reply:
Hmm... I thought about this: what if the returned stored value is defect, but not the default value?

Reply:
Why should that happen?  the Prefernces.get says:
>the value associated with key, or def if no value is associated with key, or the backingstore is inaccessible.